test/Target/cpp-calls.mlir

```diff
@@ -88,10 +88,15 @@ func @get_address_of(%arg0: i32) -> !emitc.opaque<"int32_t*"> {
   return %0 : !emitc.opaque<"int32_t*">
 }
 
+// CHECK: void emitc_constant()
 func @emitc_constant() {
-  // CHECK: int32_t [[V1]]{42};
-  %1 = "emitc.const"(){value = 42 : i32} : () -> i32
-  // CHECK: int32_t* [[V2]] = nullptr;
-  %2 = "emitc.const"(){value = "nullptr" : !emitc.opaque<"..">} : () -> !emitc.opaque<"int32_t*">
+  // CHECK: int32_t [[V0:[^ ]*]];
+  %c0 = "emitc.const"(){value = "" : !emitc.opaque<" ">} : () -> i32
```

Comment:
We should add a verfication that the op result type (i32 here) and value attribute type (!emitc.opaque<" ">) match. This is also done for the std constant op.

```c++
if (!value.getType().isa<NoneType>() && type != value.getType())
    return op.emitOpError() << "requires attribute's type (" << value.getType()
                            << ") to match op's return type (" << type << ")";
```

I'd suggest to make the types the same on the lines touched by this PR and fix the rest in a follow up.

```suggestion
  %c0 = "emitc.const"(){value = "" : !emitc.opaque<"int32_t">} : () -> !emitc.opaque<"int32_t">
```
(untested)

Reply:
I changed this to `{value = "" : i32} : () -> i32` instead.

Reply:
Sure, somehow I thought that wouldn't verify right now :)

Reply:
Right now, we don't verify anything, so you can actually do whatever comes to your mind :exploding_head: